Src/Framework/Jobs/Converter.py

```diff
@@ -13,10 +13,54 @@
 #
 # **************************************************************************
 
+import abc
+
+from Scientific.IO.NetCDF import NetCDFFile
+
 from MDANSE.Framework.Jobs.IJob import IJob
 
 class Converter(IJob):
+
+    __metaclass__ = abc.ABCMeta
     
     category = ('Converters',)
     
     ancestor = ['empty_data']
+
+    @abc.abstractmethod
+    def run_step(self,index):
+        pass
+
+    def finalize(self):
+
+        if not hasattr(self,'_trajectory'):
+            return
+
+        f = NetCDFFile(self._trajectory.filename,'a')
+
+        if 'time' in f.variables:
```

Comment:
I think it would be a good idea to wrap these if statements inside a  try finally block to ensure that the file is always closed, even if an error occurs.

Reply:
Why not

Reply:
On the other hand the if statements insure that the variables exists in the NetCDF file and so no error can occur from there.
If you prefer, according to the principle 'it is better to ask forgiveness than permission', I can replace the if by a try/except KeyError.

Reply:
I personally don't mind the if statements. If you are sure there can never be an error inside the if statements, then I guess we can leave it be. It's just that leaving a file open tends to be a really bad idea, so I think it is good to take precautions just in case.

Reply:
you right. let's add some safety guards out there